test/y2storage/clients/inst_disk_proposal_test.rb

```diff
@@ -313,12 +313,40 @@
       context "if the expert partitioner returns :back" do
         let(:result) { :back }
 
-        it "opens a new proposal dialog again with the same values" do
-          expect(Y2Storage::Dialogs::Proposal).to receive(:new).once.ordered
-            .and_return(proposal_dialog)
-          expect(Y2Storage::Dialogs::Proposal).to receive(:new).once.ordered
-            .with(proposal, devicegraph).and_return(second_proposal_dialog)
-          client.run
+        before do
+          allow(storage_manager).to receive(:staging_changed?).and_return change
+        end
+
+        context "and the devicegraph has changed" do
```

Comment:
This description is very confusing. You are checking whether `staging` is a direct copy of `probed`, which has very little to do with what I understand from the sentence "whether the devicegraph has changed".